DotNetDBF/DBFReader.cs

```diff
@@ -479,5 +479,39 @@ internal object[] NextRecord(IEnumerable<int> selectIndexes, IList<int> sortedIn
 
             return selectIndexes.Any() ? selectIndexes.Select(it => recordObjects[it]).ToArray() : recordObjects;
         }
+	
+	/// <summary>
+        /// Reads the data table from DBF from current position to end.
+        /// </summary>
+        /// <returns>The data table with all data, original types and names of columns.</returns>
+        /// <param name="dataTableName">Name for data table.</param>
+        public DataTable ReadDataTable(string dataTableName)
+        {
+            DataTable DT = new DataTable(dataTableName);
+            for (var i = 0; i < _header.FieldArray.Length; i++)
+            {
+                DBFField DF = _header.FieldArray[i];
+                string n = DF.Name;
+                DT.Columns.Add(n, DF.Type);
+                if (DF.Type == typeof(System.String))
+                    DT.Columns[n].MaxLength = DF.FieldLength;
+            }
+            object[] cDBF = NextRecord(_selectFields, _orderedSelectFields);
+            while (cDBF != null)
+            {
+                DataRow nDR = DT.NewRow();
+                for (var i = 0; i < _header.FieldArray.Length; i++)

```

Comment:
`i` isn't going to be the correct cDBF column if `_selectFields`, `_orderedSelectFields` are used.

Reply:
Thank. Now it is easier for me to use the context of your C# library/assembly. Soon I will try to write a record function from DataTable.